### PR TITLE
fix: Add module declaration files for datastore-storage-adapter

### PR DESCRIPTION
**Pull request by chintannp**
#### Description of changes
Add module declaration files for datastore-storage-adapter. This will remove the suggestedActions and error highlighting in IDE when importing `ExpoSQLiteAdapter` and `SQLiteAdapter` using the new import statements.

#### Issue #, if available
#9923
 Reported on the PR [here](https://github.com/aws-amplify/amplify-js/pull/9581#issuecomment-1132832414) by the user

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/ExpoSQLiteAdapter.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare module '@aws-amplify/datastore-storage-adapter/ExpoSQLiteAdapter';
```
[the remark this answers is not in the document]
Comment (chintannp) [reply packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/ExpoSQLiteAdapter.d.ts:1]: No, this does not get autogenerated during build time. Type declarations do get auto-generated but here we are declaring the `@aws-amplify/datastore-storage-adapter/ExpoSQLiteAdapter` as a sub module. So, that the customers can import the adapter and IDE does not complain about it not being a module.

**Review by nickarocho — APPROVED** [review]
LGTM! 👍

**Review by Samaritan1011001 — APPROVED** [review]
LGTM, thanks! Just want you to confirm these types won't get overwritten by type autogeneration if there is any.

**Comment by chintannp** [comment]
> LGTM, thanks! Just want you to confirm these types won't get overwritten by type autogeneration if there is any.

@Samaritan1011001, It won't be overwritten by type autogeneration as it only generates the types inside the `lib-esm/lib` directory. Additionally, type autogeneration does not generate type declaration for modules.

**Pull request merged**